function/dataaccess/rawls.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const {throwHttpError} = require('../errors');
+const restclient = require('./restclient');
+
+const bucketFromWorkspace = (ws) => {
+  return ws.bucketName;
+};
+
+const getBucket = async (appConfig, namespace, name, authToken) => {
+  const ws = await getWorkspace(appConfig, namespace, name, authToken);
+  return bucketFromWorkspace(ws);
+};
+
+const getWorkspace = async (appConfig, namespace, name, authToken) => {
+  const opts = appConfig || {};
+
+  if (!opts.rawlsUrl) {
+    throwHttpError(500, 'Rawls URL undefined; cannot continue.');
+  }
+
+  if (!namespace) {
+    throwHttpError(400, 'Workspace namespace is required.');
+  }
+
+  if (!name) {
+    throwHttpError(400, 'Workspace name is required.');
+  }
+
+  if (!authToken) {
+    throwHttpError(400, 'Authorization token is required.');
+  }
+
+  // query Rawls for this workspace
+  const reqOptions = {
+    uri: `${opts.rawlsUrl}/api/workspaces/${namespace}/${name}`,
+    successCodes: [200],
+    timeout: opts.timeout,
+  };
+
+  return await restclient.safeRequest(reqOptions, authToken)
+      .then((response) => {
+        // do we need "or PROJECT_OWNER" here?
```

Comment:
yes to the question here but a more correct way would be to ask sam if the user has the action (create a new action in sam, call it can_archive, assign it to owner role). The only problem is that you have workspace namespace/name but sam only knows about workspace ids. Options:
a) keep doing what you are doing and loose some flexibility
b) switch lomax apis to be in terms of workspace ids
c) rawls lookup to get workspace id from namespace/name then sam call to check action

Reply:
@dvoet yeah,  I'm definitely shortcutting here. I need to query Rawls, because I need to get the bucket info for the workspaces.  Instead of making additional call(s) to Sam, I'm just relying on what Rawls says. For the initial small-audience/whitelisted release of Lomax, I'm fine with the gross generalization of OWNER ... I could see adding a new Sam action for future releases (sending Sam the workspace ID from this Rawls call). How does that sit with you?